src/context.rs

```diff
@@ -63,7 +64,10 @@ impl<'a> QueryMatchContext<'a> {
     }
 
     pub fn get_node_text(&self, node: Node) -> &str {
-        node.utf8_text(self.file_contents).unwrap()
+        match &self.file_contents {
+            RopeOrSlice::Slice(file_contents) => node.utf8_text(file_contents).unwrap(),
+            RopeOrSlice::Rope(_) => unimplemented!(),
```

Comment:
I hadn't thought about this as a place where we were "relying" on having access to the file contents (as a string) (for "dynamically" getting the tree-sitter AST node text from inside a rule)

But that should be doable for a `Rope` too, just haven't done it yet

Maybe in theory this could also be "trait"-ified if we wanted to avoid the concrete `ropey` dependency? But not sure it matters